src/effect.h

```diff
@@ -93,6 +93,8 @@ class effect_type
         int int_decay_tick;
         time_duration int_dur_factor;
 
+        std::vector<std::string> flags;
```

Comment:
```suggestion
        std::set<std::string> flags;
```
Doesn't make sense to have multiple entries with the same value and it checking for presence of a specific value is faster. Consequently, you can load this from JSON via `flags = jo.get_tags( "flags" )`.